autowarefoundation/autoware
# Feature/gmsl multiple

## Status
**PRODUCTION / DEVELOPMENT**

## Description
A ROS driver for running multiple gmsl cameras on Nvidia PX2. The package requires DriveWorks SDK(driveworks-0.6) to be installed on the target device (e.g PX2) and drive-t186ref-linux folder    

## Steps to Test or Reproduce
1. Check the SDK and dependecy are installed and presented in these directories
  - /usr/local/driveworks
  -/usr/local/driveworks-0.6
  -/<home>/<user>/drive-t186ref-linux
or change CMakeLists.txt accordingly 

2. catkin_make

3. Connect gmsl camera(s) to a PX2 
(you may need to a parameter change in the launch file)
<param name="selector_mask" value="11111111" type="str" />
to reflect the actual camera connection
port A|B|C      
e.g 5 cameras on port A (4) and B(1)
<param name="selector_mask" value="11110001" type="str" />

```
roslaunch gmsl_camera multi_gmsl.launch
```
3. rosrun rqt_gui rqt_gui
topic name "gmsl_image_raw_0_0" for example
